flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "Flake for Plutus";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=7d71001b796340b219d1bfa8552c81995017544a";
+    haskell-nix.url = "github:input-output-hk/haskell.nix?rev=87084d65a476cc826a0e8c5d281d494254f5bc7a";
+    flake-utils.url = "github:numtide/flake-utils?rev=b543720b25df6ffdfcf9227afafc5b8c1fabfae8";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, flake-utils, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        topLevel = import ./. {
+          inherit system;
+          haskellNixOverlays = [ haskell-nix.overlay ];
+        };
+
+        inherit (topLevel) pkgs plutus ownOverlays;
+        inherit (plutus) haskell iohkNix;
+        inherit (plutus.lib) buildPursPackage buildNodeModules filterNpm gitignore-nix;
```

Comment:
most of these are unused, why are they here? Just for experimentation?

Reply:
i cleaned those up.